src/Main/Main.cc

```diff
@@ -86,6 +86,12 @@ int main(int argc, char* argv[]) {
             }
         }
 
+        if (settings.controller_deployment) {
+            setenv("CONTROLLER_DEPLOYMENT", "controller", 1);
```

Comment:
You should not be setting and then reading an environment variable in order to convey this information from one part of the code to a different part of the code. This should be done internally. We already have a mechanism for using static variables on the Session class to save certain settings (various timeouts) which can then be used inside session functions. Please use the same method to save this value.

Reply:
Thank you so much for your patience reviewing my code! I fixed it according to your requirements.